src/Avalonia.FuncUI/DSL/ItemsControl.fs

```diff
@@ -15,19 +20,64 @@ module ItemsControl =
 
     type ItemsControl with
 
+        static member displayMemberBinding<'t when 't :> ItemsControl>(value: IBinding) : IAttr<'t> =
+            AttrBuilder<'t>.CreateProperty<IBinding>(ItemsControl.DisplayMemberBindingProperty, value, ValueNone)
+
         static member viewItems<'t when 't :> ItemsControl>(views: IView list) : IAttr<'t> =
             let getter : ('t -> obj) = (fun control -> control.Items :> obj)
             AttrBuilder<'t>.CreateContentMultiple("Items", ValueSome getter, ValueNone, views)
 
         static member dataItems<'t when 't :> ItemsControl>(data: IEnumerable) : IAttr<'t> =
             AttrBuilder<'t>.CreateProperty<IEnumerable>(ItemsControl.ItemsSourceProperty, data, ValueNone)
 
+        static member itemContainerTheme<'t when 't :> ItemsControl>(value: ControlTheme) : IAttr<'t> =
+            AttrBuilder<'t>.CreateProperty<ControlTheme>(ItemsControl.ItemContainerThemeProperty, value, ValueNone)
+
         static member itemsPanel<'t when 't :> ItemsControl>(value: ITemplate<Panel>) : IAttr<'t> =
             AttrBuilder<'t>.CreateProperty<ITemplate<Panel>>(ItemsControl.ItemsPanelProperty, value, ValueNone)
 
         static member itemTemplate<'t when 't :> ItemsControl>(value: IDataTemplate) : IAttr<'t> =
             AttrBuilder<'t>.CreateProperty<IDataTemplate>(ItemsControl.ItemTemplateProperty, value, ValueNone)
 
+        static member onContainerPrepared<'t when 't :> ItemsControl>(func: ('t * ContainerPreparedEventArgs) -> unit, ?subPatchOptions) =
+            let name = nameof Unchecked.defaultof<'t>.ContainerPrepared
+            let factory: AvaloniaObject * ('t * ContainerPreparedEventArgs -> unit) * CancellationToken -> unit =
+                fun (control, func, token) ->
+                    let control = control :?> 't
+                    let handler = EventHandler<ContainerPreparedEventArgs>(fun s e -> func(s :?> 't, e))
+                    let event = control.ContainerPrepared
+
+                    event.AddHandler(handler)
+                    token.Register(fun _ -> event.RemoveHandler(handler)) |> ignore
+
+            AttrBuilder<'t>.CreateSubscription<'t * ContainerPreparedEventArgs>(name, factory, func, ?subPatchOptions = subPatchOptions)
```

Comment:
Do subscriptions expire here when a new handler is added (say because SubPatchOptions change)?

Reply:
Based on the implementation of patchSubscription, it seems that if we register the unsubscription process with CancellationToken, it will be executed properly, but I'm not entirely sure. 

https://github.com/SilkyFowl/Avalonia.FuncUI/blob/7107231775827b79546c5be38deadc4ee8e8ca46/src/Avalonia.FuncUI/VirtualDom/VirtualDom.Patcher.fs#L21-L49

I'm thinking of adding some test code to verify this.

Reply:
Once this is verified we can merge IMO